src-web/components/modals/AnsibleAutomationModal/AnsibleAutomationModal.js

```diff
@@ -555,27 +563,27 @@ export class AnsibleAutomationModal extends React.Component {
   getAnsibleConnection = (ansCredentials, credentialNameNew, type) => {
     const { credentialName:credentialNameOld, credentialNS } = this.state
     const targetCredential = _.find(ansCredentials, ['name', (credentialNameNew || credentialNameOld)])
+    let towerURL='', token=''
     if (targetCredential && targetCredential.namespace
       && targetCredential.host && targetCredential.token) {
+      towerURL = targetCredential.host
+      token = targetCredential.token
       if (credentialNS !== targetCredential.namespace) {
         this.setState({
           credentialNS: targetCredential.namespace,
-          towerURL:targetCredential.host,
+          towerURL,
         })
       }
       if (type === 'edit') {
         this.setState({
           ansibleConnection: {
-            towerURL: targetCredential.host,
-            token: targetCredential.token
+            towerURL,
+            token,
           }
         })
       }
     }
-    return {
-      towerURL: targetCredential.host,
-      token: targetCredential.token
-    }
+    return { towerURL, token }
```

Comment:
Go through the whole logic and looks good to me. Thanks for fixing this line and other corner cases handling